datafusion/expr/src/logical_plan/ddl.rs

```diff
@@ -192,11 +193,12 @@ impl DdlStatement {
 
 /// Creates an external table.
 #[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
```

Comment:
Why make it non exhaustive? That might make it harder to write `match` statements where all the fields are matched (and the compiler tells you when new ones are added)

Reply:
> Why make it non exhaustive?

To disallow public direct construction. If we allow public direct construction, we can't enforce invariants in the factory method, since it may or may not get called.

> That might make it harder to write match statements where all the fields are matched (and the compiler tells you when new ones are added)

💯  and i think this is important use-case. that's why those new `into_fields()` are added